microprofile/config/src/test/java/io/helidon/microprofile/config/ConfigBeanDescriptorTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.config;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import io.helidon.config.mp.MpConfigSources;
+import io.helidon.microprofile.config.testsubjects.TestConfiguredBean;
+
+import jakarta.enterprise.inject.spi.AnnotatedType;
+import jakarta.enterprise.inject.spi.InjectionPoint;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperties;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Lower level testing for {@link io.helidon.microprofile.config.ConfigBeanDescriptor}.
+ */
+class ConfigBeanDescriptorTest {
+
+    private static MutableMpTest.MutableSource source;
+    private static Config config;
+    private static io.helidon.config.Config emptyConfig;
+
+    @BeforeAll
+    static void initClass() {
+        config = createTestConfiguredBeanConfig();
+
+        // we need to ensure empty config is initialized before running other tests,
+        // as this messes up the mapping service counter
+        emptyConfig = io.helidon.config.Config.empty();
+
+        source = new MutableMpTest.MutableSource("initial");
+
+        // register config
+        ConfigProviderResolver configProvider = ConfigProviderResolver.instance();
+        configProvider.registerConfig(config, Thread.currentThread().getContextClassLoader());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void testProduceListFields() throws Exception {
+        AnnotatedType<?> annotatedType = mock(AnnotatedType.class);
+        when(annotatedType.getJavaClass()).thenReturn((Class) TestConfiguredBean.class);
+
+        Field strListField = Objects.requireNonNull(TestConfiguredBean.class.getField("strList"));
```

Comment:
`Objects.requireNonNull` is not the right choice here. If you want to check for null, use an assertion.